zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractPlugin.java

```diff
@@ -1142,4 +1146,191 @@ public int getWascId() {
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setRisk(int risk) {
+            super.setRisk(risk);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setConfidence(int confidence) {
+            super.setConfidence(confidence);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setDescription(String description) {
+            super.setDescription(description);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setUri(String uri) {
+            super.setUri(uri);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setParam(String param) {
+            super.setParam(param);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setAttack(String attack) {
+            super.setAttack(attack);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setOtherInfo(String otherInfo) {
+            super.setOtherInfo(otherInfo);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setSolution(String solution) {
+            super.setSolution(solution);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setReference(String reference) {
+            super.setReference(reference);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setEvidence(String evidence) {
+            super.setEvidence(evidence);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setCweId(int cweId) {
+            super.setCweId(cweId);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setWascId(int wascId) {
+            super.setWascId(wascId);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setMessage(HttpMessage message) {
+            super.setMessage(message);
+            messageSet = message != null;
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setSourceHistoryId(int sourceHistoryId) {
+            super.setSourceHistoryId(sourceHistoryId);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setHistoryRef(HistoryReference historyRef) {
+            super.setHistoryRef(historyRef);
+            return this;
+        }
+
+        @Override
+        public AlertBuilder setSource(Source source) {
+            super.setSource(source);
+            return this;
+        }
+
+        /**
+         * Raises the alert with specified data.
+         *
+         * @throws IllegalStateException if the HTTP message was not set.
+         */
+        public void raise() {
+            if (!messageSet) {
+                throw new IllegalStateException(
+                        "A HTTP message must be set before raising the alert.");
```

Comment:
Should this include source and plugin ID to assist troubleshooting?

Reply:
Can add, but the stacktrace will already include the scanner/class.

Reply:
Okay don’t worry about it.